ingestion/ingestion_server/ingestion_file_upload_rest_api.py

```diff
@@ -458,57 +456,68 @@ def file_sharing():
             response = build_file_share_error_response(asset_id, HTTPStatus.BAD_REQUEST.phrase)
             return json.dumps(response), HTTPStatus.BAD_REQUEST
 
+        if not revoke_users:
+            # TODO: update the users list in OPA
+            pass
+
+        # TODO: add receivers to users list in OPA
         server_urls_instance = ServerUrls().get_instance()
         get_return_obj = RestClientApis.http_get_and_check_success(
             server_urls_instance.key_server_single_asset_url.format(asset_id))
 
-        if get_return_obj.success:
-            symmetric_key = get_return_obj.to_dict()['json']['response']['key']['key']
-            db_core = MainDb.get_core_db_instance()
-            fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
-
-            for person in receiver:
-                try:
-                    # finds the receivers public key file for symmetric key encryption
-                    user_pubkey = fs.find({'metadata.owner': person, 'metadata.type': 'public key'})
-                    if user_pubkey.count():
-                        for name in user_pubkey:
-                            fname = name.filename
-                        src_file_path = os.path.join(IngestionGlobals().data_dir, fname)
-
-                        # RSA asymmetric encryption algorithm used
-                        with open(src_file_path) as data:
-                            public_key = RSA.importKey(data.read())
-                            cipher = PKCS1_OAEP.new(public_key)
-                            cipher_text = cipher.encrypt(symmetric_key.encode("utf-8"))
-
-                        # cipher_text stored as hex string in json response
-                        file_share_response = build_file_share_response(asset_id, cipher_text.hex())
-                        response_dict[person] = file_share_response
-                    else:
-                        raise Exception('Public key does not exists')
-
-                except Exception as e:
-                    message = str(e)
-                    response_dict[person] = build_file_share_error_response(asset_id, message)
-                    code = HTTPStatus.INTERNAL_SERVER_ERROR
-
-            resp = json.dumps(response_dict)
-            return resp, code
+        if not get_return_obj.success:
+            raise Exception('Key Server problem')
 
-        else:
-            raise Exception("Key Server problem")
+        symmetric_key = get_return_obj.to_dict()['json']['response']['key']['key']
+        db_core = MainDb.get_core_db_instance()
+        fs = gridfs.GridFSBucket(db_core.get_magen_mdb())
+
+        for person in receivers:
+            try:
+                with db.connect(config.DEV_DB_NAME) as db_instance:
+                    result = user_model.UserModel.select_by_email(db_instance, person)
+                # Checking if the person exists or not
+                if not result.count:
+                    raise Exception('User ' + person + ' does not exist')
```

Comment:
Is this situation really an *exceptional*? Can a person pass an e-mail of a person that Magen doesn't know about? If so, then we should not handle this situation as an exception - we should process it normally.

I would have enhanced function ```build_file_share_error_response()``` to be more general and would called it here and in your ```except``` block. I think a good idea to allow passing ```**kwargs``` to ```build_file_share_error_response()``` and act appropriately on passing either exception object or general error message